server/router/profile.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+
+const Profile = require('../models/profileModel')
+
+const router = new express.Router();
+
+//An endpoint to fetch all of the user profiles
+router.get('/profiles', async (req,res) => {
+    try {
+        const profiles = await Profile.find();
+        return res.json(profiles)
+    } catch(err) {
+        console.log(`Exception caught --------> ${err}`)
+        return res.status(500).send(err);
+    }
+})
+
+
+//To register a new user
+router.post('/signup', async (req,res) => {
+    try {
+        const profile = new Profile(req.body);
+        await profile.save();
+        return res.status(201).json({"type": "SUCCESS","message":"Voter registered"})
+    } catch(e) {
+        console.log(`Exception caught --------> ${e}`)
+        return res.status(500).send(e);
+    }
+})
+
+//To authenticate a user
+router.post('/login', async (req,res) => {
+    try {
+        if(req.body.username && req.body.password) {
+            const profileObj = await Profile.findOne(req.body);
+            console.log(profileObj);
+            if(profileObj && profileObj.password === req.body.password) {
+                console.log("Successfully Authenticated")
+                return res.status(200).json({"type": "SUCCESS","message":"User logged in"})
+            } else {
+                console.log("Incorrect password")
```

Comment:
It shouldn't say which field is incorrect. Makes it easier for the attacker to figure out which one to brute force.

Reply:
We should also add captcha.

Reply:
Since it's in the server side, the attacker shouldn't be able to see it ideally. But yes it's always safe to have these kinds of things removed from the code. Have done so: https://github.com/meyeralex31/csci550/commit/b4e0caddb2f3b7a64bd83dc18889419fc1f4faea